pyledger/memory_ledger.py

```
@@ -0,0 +1,325 @@
+        """Modifies an existing ledger entry in the local ledger.
+
+        Args:
+            entry (pd.DataFrame): DataFrame with ledger entry in pyledger schema.
+
+        Raises:
+            ValueError: If the entry does not have a unique 'id' across all rows.
+            ValueError: If the ledger entry with the specified 'id' does not exist in the ledger.
+        """
+        if entry["id"].nunique() != 1:
+            raise ValueError(
+                "Id needs to be unique and present in all rows of a collective booking."
+            )
+
+        ledger_id = str(entry["id"].iat[0])
+        if ledger_id not in self._ledger["id"].values:
+            raise ValueError(f"Ledger entry with id '{ledger_id}' not found.")
+
+        self._ledger = self.standardize_ledger(pd.concat(
+            [self._ledger[self._ledger["id"] != ledger_id], entry],
+        ))
+
+    def delete_ledger_entry(self, id: str) -> None:
+        """Deletes a ledger entry from the local ledger.
+
+        Args:
+            id (str): The Id of the ledger entry to be deleted.
+        """
+        if id in self._ledger["id"].values:
+            self._ledger = self._ledger[self._ledger["id"] != id]
+
+    def mirror_ledger(self, target: pd.DataFrame, delete: bool = False):
+        """Synchronizes ledger entries in the memory with a desired target state
+        provided as a DataFrame.
+
+        Args:
+            target (pd.DataFrame): DataFrame with ledger entries in the pyledger format.
+            delete (bool, optional): If True, deletes ledger entries in the memory that are not
+                                    present in the target DataFrame.
+        """
+        target_df = self.standardize_ledger(target)
+
+        if delete:
+            self._ledger = target_df
+        else:
+            self._ledger = self.standardize_ledger(
+                pd.concat([self._ledger, target_df]).drop_duplicates()
+            )
```

The current logic only drops 'identical' duplicates. Our implementation in cashctrl_ledger is smarter: it drops duplicates that are financially equivalent.

There is no need for action at this time. The two implementations will automatically align as part of #25 which is scheduled for completion in October.

---

Since the current logic is not correct, we could raise an error until #25 is implemented:
```python
            raise NotImplementedError(
                "mirror_ledger(..., delete=False) will be implemented as part of #25.")
```
Instead of:
```python
            self._ledger = self.standardize_ledger(
                pd.concat([self._ledger, target_df]).drop_duplicates()
            )
```